### ISSUE TITLE
Errors in development reference dist/server_dev.js instead of project source code

### ISSUE THREAD

**Issue by arsanjea** [reactql/kit#93]
While using the latest version of reactql, errors in development mode are referencing the compiled dev source code (dist/server_dev.js), making it challenging to debug my project.

**Comment by noeljackson** [comment]
The source map is not engaging properly to give you the mapped location of the errors?

**Comment by arsanjea** [comment]
Nope. Not sure why. I haven't made any change to the kit and config folders...just to the src folder

**Comment by arsanjea** [comment]
Also, chrome://inspect does not seem to stop at breakpoint or to allow me to view source files it is referring to (although source maps are enabled). And that's on two very distinct devices.

**Comment by arsanjea** [comment]
@leebenson any idea as to what's happening?

**Comment by nate0001** [comment]
I've seen this as well. Sometimes errors reference dist/ code and it makes it hard to debug.

**Comment by leebenson** [comment]
Is this something that was confirmed working in previous versions? AFAIK, the server (both dev and production) was never targeted for source mapping. I've tried `devtool: 'eval'` and all other source variants, and the best I've been able to get so far is this:

> webpack:///./src/app.js?:21
  throw new Error('bla');
  ^
> Error: bla
    at ServerConfig._config2.default.getKoaApp.app [as koaAppFunc] (webpack:///./src/app.js?:21:9)
    at eval (webpack:///./kit/entry/server.js?:486:20)
    at Object.<anonymous> (/Users/leebenson/dev/sandbox/reactql-fonts/dist/server_dev.js:135:1)

... which is _sort of_ in the ballpark of where the error is throwing (it has the correct source file, but the line numbers are) off; the stack trace is entirely obfuscated.

If someone has the time to debug and propose a fix for Node bundling with source maps, I'd be grateful for a PR. Otherwise, I'll put this on the to-do to investigate when I can.

**Comment by arsanjea** [comment]
It was actually pretty easy to solve (but maybe I am missing the point)
You just need to add source-map-support as a devDependency and add the following at the top of your code: 

```
if(process.env.NODE_ENV === 'development') {
  require('source-map-support').install();
}
```

**Comment by leebenson** [comment]
@arsanjea - would you mind sending a PR for this? adding this locally doesn't yield any change, so I'd be interested to see how you've done it.

**Comment by nate0001** [comment]
There's also this babel plugin. https://www.npmjs.com/package/babel-plugin-source-map-support

Not sure what would be the better solution.

**Comment by arsanjea** [comment]
Sure - traveling today, but I’ll take care of it tomorrow.
Again, not sure it’s the best solution.

Sincerely,
Jean-Marc Arsan

C +1 (310) 435-9983

**Comment by leebenson** [comment]
Apologies for the generic response, but I've released a new [v3.0 kit](https://github.com/leebenson/reactql/releases/tag/3.0.0), which has been re-written from the ground up.

It bumps the ReactQL stack to Webpack 4, React 16.4, Apollo Client 2.0, and adds Styled Components, full SSR (even in development), and has been re-written in Typescript. It's leaner, faster and has been brought completely up-to-date.

If you're still using ReactQL (and I appreciate you may not be, given the time lag in my response), I'd recommend checking out the new kit to see if your issue has been answered. If not, please feel free to post a new issue at https://github.com/leebenson/reactql
